specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2024-04-01/registries.json

```diff
@@ -710,6 +718,34 @@
       },
       "additionalProperties": false
     },
+    "ManagedResourceGroupSettings": {
+      "description": "Managed resource group settings",
+      "type": "object",
+      "properties": {
+        "assignedIdentities": {
+          "description": "List of assigned identities for the managed resource group",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ManagedResourceGroupAssignedIdentities"
+          },
+          "x-nullable": true,
+          "x-ms-identifiers": []
+        }
+      },
+      "additionalProperties": false
+    },
+    "ManagedResourceGroupAssignedIdentities": {
+      "description": "Details for managed resource group assigned identities.",
+      "type": "object",
+      "properties": {
+        "principalId": {
```

Comment:
>principalId [](http://example.com/codeflow?start=9&length=11)

how do you use this field ? Is this provided by the user ? If you hard code these you will face issues if the tenant of the subscription changes

Reply:
This is a field to track/record any tenants with access to the MachineLearning Registry's storage resources.
They are added/removed when the user requests access to the resources in the MLRegistries.
The user should be charged for ML Registries and resources already, this tracking won't generate any additional cost.
It is only used for security checks.

Reply:
Has this been signed off by the RBAC team ?[https://apps.preview.powerapps.com/play/e/839eace6-59ab-4243-97ec-a5b8fcc104e4/a/241165b8-9173-4e16-9876-a1e2138aa130?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47](https://apps.preview.powerapps.com/play/e/839eace6-59ab-4243-97ec-a5b8fcc104e4/a/241165b8-9173-4e16-9876-a1e2138aa130?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)


This is not a good mechanism since subscriptions can and do move across tenants causing Principalsids to change. Custom RBAC checks like this are highly discouraged in ARM

Reply:
We'll handle the subscription move by blanking those fields when it happens.

Reply:
Could you just book a slot in the arm api office hours so that we can close since this seems like it needs a discussion

Reply:
sure